policy-controller/k8s/index/src/index.rs

```diff
@@ -212,14 +213,18 @@ impl kubert::index::IndexNamespacedResource<k8s::policy::Server> for Index {
 
         for (namespace, Ns { added, removed }) in updates_by_ns.into_iter() {
             if added.is_empty() {
+                // If there are no live resources in the namespace, we do not
+                // want to create a default namespace instance, we just want to
+                // clear out all resources for the namespace (and then drop the
+                // whole namespace, if necessary).
```

Comment:
thanks for this comment, it's much more obvious what's going on here now. it could be worth explaining why it's okay to clear _everything_ rather than just the resources that were removed?

Reply:
Yep reading through this again from yeserday this comment makes things clearer.